### ISSUE TITLE
Breaks when using the proxy option

### ISSUE THREAD

**Issue by mfeckie** [healthsparq/ember-fountainhead#34]
Nice work on the leap to 3.0!

When starting ember using `ember server --proxy=http://localhost:3000` this interferes with the middleware added by fountainhead.  Not sure that we can have serverMiddleware and proxy at the same time 😢 

Workaround that I'm using for now.

`ember generate http-proxy api-v1`

`app/server/proxies/api-v1.js`

```js
/*eslint-env node*/
var proxyPath = '/api';

module.exports = function(app) {
  // For options, see:
  // https://github.com/nodejitsu/node-http-proxy
  var proxy = require('http-proxy').createProxyServer({});

  proxy.on('error', function(err, req) {
    console.error(err, req.url);
  });

  app.use(proxyPath, function(req, res /*, next */){
    req.url = proxyPath + req.url;
    proxy.web(req, res, { target: {
      host: 'localhost',
      port: process.env['API_PORT'] || 5000
    }});
  });
};
```

So this basically intercepts the api requests and means we can revert to using `ember server` without the proxy flag.

**Comment by DHedgecock** [comment]
Hey @mfeckie , you wouldn't happen to have an old set of generated docs in your /public directory would you?

I ran v3 with an addon and an app with proxy and live reload worked with both.

The live reload wasn't working for a different app I had using an older version of Fountainhead until I deleted the old generated docs from the /public directory.

:crossed_fingers: that's breaking your setup. If it is, super sorry for the wasted time. I did not think of this ahead of time.

**Comment by mfeckie** [comment]
I'll take a poke later.

Please don't apologise!  I'm loving what you folks are doing and deeply appreciate the library!  Open source is so often thankless and I hate piling on issues 😄

**Comment by mfeckie** [comment]
Ah, OK, so I think we may be talking at cross purposes.

The docs get generated just fine, they just aren't served correctly via the app.

E.G.  Visit `http://localhost/#/api` <- (We're using hash `locationType`)

This bring us to the ember route correctly, then we have

```
GET http://localhost:4200/docs/meta.json 404 (Not Found)
```

This request is being forwarded straight to the backend and not being served by the express.js static route inserted by fountainhead.

**Comment by DHedgecock** [comment]
Yep, nevermind my earlier comment, you are totally right. When running serve with --proxy, the CLI server middleware is called first and adds this middleware:
`app.use((req, res) => proxy.web(req, res));`, it intercepts ALL requests, so the middleware from this addon never gets called. _Sigh_ Life is never easy.

The only ray of sunshine that I found was that generated http-mocks middleware are registered before the CLI proxy is, so it should work to do this:

`ember g http-mock docs`

```javascript
// server/mocks/docs.js

module.exports = function(app) {
  var express = require('express');
  app.use('/docs', express.static('docs'));
};
```

Basically the same thing you're doing as a workaround, but with using a mock using -proxy should work again.

Alternatively, depending on their setup, people could add the express static middleware directly to their proxied servers (we had an ooooooold middleware on our server set up like this that was masking the issue).

Basically if you have:

- `/root/code/ember-app/docs`
- `/root/code/server`

You could:

```javascript
// server/router.js

// server express app logic...

if (process.env.NODE_ENV !== 'production') {
  app.use('/docs', express.static(path.resolve('..', 'ember-app', 'docs')));
}
```

But I'd like to keep the addon setup from creeping out like that...


At this point I don't see a way to 'fix' this, which is a bummer. I'm going to update the docs with an explanation of the issue and examples of your workaround and updating the backend server and then close this ticket. If anyone ever thinks of a way to 'fix' this, or if CLI updates to let addons register middleware before the CLI proxy, we can reopon it.


Big thanks for logging the ticket and thinking of the workaround. It's much appreciated.

**Comment by DHedgecock** [comment]
JK JK. Great news, this should be fixed 100% in 3.0.2 by configuring this addon to run before the proxy server addon. You should be able to delete that http-proxy entirely and go back to using -proxy.

**Comment by mfeckie** [comment]
Fixed and confirmed :)  Great work human(s)
